qiskit/scheduler/methods/basic.py

```diff
@@ -93,7 +93,8 @@ def as_late_as_possible(circuit: QuantumCircuit,
     """
     sched = Schedule(name=circuit.name)
     # Align channel end times.
-    circuit.barrier()
+    circuit_barrier = circuit.copy()
```

Comment:
I think this is fine. There is no need to rename though. Could just do
```suggestion
    circuit = circuit.copy()
```

Reply:
Actually I think what Thomas originally suggested can be done with this. Can you try this? I think it's faster than a deepcopy.
```suggestion
    circuit = circuit + QuantumCircuit(*circuit.qregs)
```